src/diffusers/schedulers/scheduling_ddpm.py

```diff
@@ -26,6 +26,17 @@
 from .scheduling_utils import SchedulerMixin
 
 
+def expand_to_shape(input, timesteps, shape, device):
+    """
+    Helper indexes a 1D tensor `input` using a 1D index tensor `timesteps`, then reshapes the result to broadcast
+    nicely with `shape`. Useful for parallelizing operations over `shape[0]` number of diffusion steps at once.
+    """
+    out = torch.gather(input.to(device), 0, timesteps.to(device))
+    reshape = [shape[0]] + [1] * (len(shape) - 1)
+    out = out.reshape(*reshape)
+    return out
+
+
```

Comment:
How do we feel about moving this to `scheduling_utils.py`? Maybe `get_alpha_sigma` as well.

Reply:
I'm good with this. I had this as a TODO in my mind. Could also be made more elegant, but wasn't 100% sure how yet.

Reply:
My only concern with these two is
1. `expand_to_shape` would be the only function like this. It's okay to start the trend.
2. `get_alpha_sigma` won't work with many of the schedulers, so I'm okay with leaving it in the ones that use v-prediction for now.